CommunityToolkit.Tooling.SampleGen/ToolkitSampleMetadataGenerator.Documentation.cs

```diff
@@ -150,6 +156,7 @@ private ImmutableArray<ToolkitFrontMatter> GatherDocumentFrontMatter(SourceProdu
 
                 // Get the filepath we need to be able to load the markdown file in sample app.
                 var filepath = file.Path.Split(new string[] { @"\components\", "/components/", @"\tooling\", "/tooling/" }, StringSplitOptions.RemoveEmptyEntries).LastOrDefault();
+                var iconpath = filepath.Replace(Path.GetFileName(filepath), icon);
```

Comment:
@Arlodotexe Not sure if there's a cleaner way to do this - also related to line 214

Reply:
Think this would make the icon path relative to the markdown file, rather than the relative to the root of the component's sample project. The latter prevents icons from breaking when the markdown files are moved around.

Let's use the .md path to get the path of the containing sample project, then combine it with the relative icon path provided in the front matter.

Reply:
@arlodotexe How would we get the path of the containing sample project - couldn't fogure that out since the .md path is already relative?

Reply:
Dug in a bit more. We currently replace the markdown file name with the icon path, making the icon path relative to the folder the markdown is in. That works fine when everything is in root, but putting anything in a subfolder would require usage of a parent path indicator (e.g. `./../`) which may not work end-to-end without resolving the path.

The Markdown file is provided to us as an absolute path, such as:
```
D:\source\uwp\lib\Labs-Windows\components\CanvasLayout\samples\CanvasLayout.md
```

To get a `filepath` we can use for `ms-appx:///`, we shorten it to:
```
CanvasLayout\samples\CanvasLayout.md
```

So we'd pull out the substring `CanvasLayout\samples\` and append the icon path instead of replacing the file name.

Reply:
Updated.. I think this should work? Let me know if you see a cleaner way.